test/integration/ethereum/flashMintLeveraged.spec.ts

```diff
@@ -0,0 +1,466 @@
+                  } else {
+                    inputTokenBalance = await inputToken.balanceOf(owner.address);
+                    subjectMaxAmountIn = inputTokenBalance;
+                    await inputToken.approve(flashMintLeveraged.address, subjectMaxAmountIn);
+                    subjectInputToken = inputToken.address;
+                  }
+                  subjectSetToken = setToken.address;
+                });
+
+                async function subject() {
+                  if (inputTokenName == "ETH") {
+                    return flashMintLeveraged.issueExactSetFromETH(
+                      subjectSetToken,
+                      subjectSetAmount,
+                      swapDataDebtToCollateral,
+                      swapDataInputToken ,
+                      { value: subjectMaxAmountIn },
+                    );
+                  }
+                  return flashMintLeveraged.issueExactSetFromERC20(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    subjectInputToken,
+                    subjectMaxAmountIn,
+                    swapDataDebtToCollateral,
+                    swapDataInputToken ,
+                );
+                }
+
+                async function subjectQuote() {
+                  return flashMintLeveraged.callStatic.getIssueExactSet(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    swapDataDebtToCollateral,
+                    swapDataInputToken ,
+                );
+                }
+
+
+                it("should issue the correct amount of tokens", async () => {
+                  const setBalancebefore = await setToken.balanceOf(owner.address);
+                  await subject();
+                  const setBalanceAfter = await setToken.balanceOf(owner.address);
+                  const setObtained = setBalanceAfter.sub(setBalancebefore);
+                  expect(setObtained).to.eq(subjectSetAmount);
+                });
+
+                it("should spend less than specified max amount", async () => {
+                  const inputBalanceBefore =
+                    inputTokenName == "ETH"
+                      ? await owner.wallet.getBalance()
+                      : await inputToken.balanceOf(owner.address);
+                  await subject();
+                  const inputBalanceAfter =
+                    inputTokenName == "ETH"
+                      ? await owner.wallet.getBalance()
+                      : await inputToken.balanceOf(owner.address);
+                  const inputSpent = inputBalanceBefore.sub(inputBalanceAfter);
+                  expect(inputSpent.gt(0)).to.be.true;
+                  expect(inputSpent.lte(subjectMaxAmountIn)).to.be.true;
+                });
+
+                it("should quote the correct input amount", async () => {
+                  const inputBalanceBefore =
+                    inputTokenName == "ETH"
```

Comment:
small nits to make `==` into `===` and `!=` into `!==`

Reply:
resolved in:
https://github.com/IndexCoop/index-coop-smart-contracts/pull/143/commits/15b7fac84f9f9ffcdbc1ae0ca3c90bf27af95a36